src/test/lib/storage/index/partial_hash/partial_hash_index_test.cpp

```diff
@@ -0,0 +1,455 @@
+    pmr_vector<bool> null_values_1 = {false, false, true, false, false, false, false, false};
+    values2 = {"hello", "delta", "funny", "names", "nullptr", "paper", "clock", "inbox"};
+    pmr_vector<bool> null_values_2 = {false, false, false, false, true, false, false, false};
+    segment1 = std::make_shared<ValueSegment<pmr_string>>(std::move(values1), std::move(null_values_1));
+    segment2 = std::make_shared<ValueSegment<pmr_string>>(std::move(values2), std::move(null_values_2));
+
+    Segments segments1 = {segment1};
+    Segments segments2 = {segment2};
+
+    table->append_chunk(segments1);
+    table->append_chunk(segments2);
+
+    std::vector<std::pair<ChunkID, std::shared_ptr<Chunk>>> chunks_to_index;
+    chunks_to_index.push_back(std::make_pair(ChunkID{0}, table->get_chunk(ChunkID{0})));
+    chunks_to_index.push_back(std::make_pair(ChunkID{1}, table->get_chunk(ChunkID{1})));
+
+    index = std::make_shared<PartialHashIndex>(chunks_to_index, ColumnID{0});
+
+    index_map = &(std::dynamic_pointer_cast<PartialHashIndexImpl<pmr_string>>(index->_impl)->_map);
+  }
+
+  pmr_vector<pmr_string> values1;
+  pmr_vector<pmr_string> values2;
+  std::shared_ptr<ValueSegment<pmr_string>> segment1 = nullptr;
+  std::shared_ptr<ValueSegment<pmr_string>> segment2 = nullptr;
+  std::shared_ptr<Table> table = nullptr;
+  std::shared_ptr<PartialHashIndex> index = nullptr;
+
+  /**
+   * Use pointers to inner data structures of PartialHashIndex in order to bypass the
+   * private scope. Since the variable is set in setup() references are not possible.
+   */
+  tsl::robin_map<pmr_string, std::vector<RowID>>* index_map = nullptr;
+};
+
+TEST_F(PartialHashIndexTest, Type) { EXPECT_EQ(index->type(), TableIndexType::PartialHash); }
+
+TEST_F(PartialHashIndexTest, IndexCoverage) {
+  EXPECT_EQ(index->get_indexed_chunk_ids(), (std::set<ChunkID>{ChunkID{0}, ChunkID{1}}));
+
+  EXPECT_TRUE(index->is_index_for(ColumnID{0}));
+  EXPECT_FALSE(index->is_index_for(ColumnID{1}));
+}
+
+TEST_F(PartialHashIndexTest, EmptyInitialization) {
+  EXPECT_THROW(PartialHashIndex(std::vector<std::pair<ChunkID, std::shared_ptr<Chunk>>>(), ColumnID{0}),
+               std::logic_error);
+  auto empty_index = PartialHashIndex(DataType::String, ColumnID{0});
+
+  EXPECT_EQ(empty_index.cbegin(), empty_index.cend());
+  EXPECT_EQ(empty_index.null_cbegin(), empty_index.null_cend());
+
+  EXPECT_EQ(empty_index.range_equals("any").first, empty_index.cend());
+  EXPECT_EQ(empty_index.range_equals("any").second, empty_index.cend());
+
+  EXPECT_EQ(empty_index.get_indexed_chunk_ids().size(), 0);
+
+  EXPECT_TRUE(empty_index.is_index_for(ColumnID{0}));
+  EXPECT_FALSE(empty_index.is_index_for(ColumnID{1}));
+
+  EXPECT_EQ(empty_index.cbegin(), empty_index.cend());
+  EXPECT_TRUE(empty_index.cbegin().operator==(empty_index.cbegin()));
+  EXPECT_FALSE(empty_index.cbegin().operator!=(empty_index.cbegin()));
+}
+
+TEST_F(PartialHashIndexTest, MapInitialization) {
```

Comment:
In the original PR it was suggested to extract the tests for each string into a method in order to make it more compact.
Another opinion was to keep the current implementation due to readability and understandability.
Which option do you prefer?

Reply:
I prefer the second approach, i.e., having it more explicit.